be/test/io/s3_input_stream_test.cpp

```diff
@@ -40,21 +39,31 @@ class S3InputStreamTest : public testing::Test {
 
     static void put_object(const std::string& object_content);
 
-    void SetUp() override {}
+    void SetUp() override {
+        if (!env_util::is_running_on_ec2()) {
+            GTEST_SKIP() << "not running on ec2";
+        }
+    }
 
     void TearDown() override {}
 
     std::unique_ptr<S3InputStream> new_random_access_file();
 };
 
 void S3InputStreamTest::SetUpTestCase() {
+    if (!env_util::is_running_on_ec2()) {
```

Comment:
Why check this? S3InputStream/S3OutpuStream can be used to read and write any service compatible with S3 protocol, including OSS.

Reply:
Actually, I'm using OSS as the endpoint when running this test on my local machine.

Reply:
You can keep the `with-aws` flag in run-ut.sh, and if it's false, you can run the unit tests with flag '--gtest-filter=-*S3*'.

Reply:
OK. I'll fix for that.

Reply:
Whether the tests are running on EC2 does not matter, what really matters is whether you have a correct S3 configuration in the file `be.conf`. For security reasons, we don't provide an accessible S3 endpoint in `be.conf` by default, that's why the S3 tests are disabled by default. 
But for developers who have configured a valid S3 endpoint and AK/SK, they can run these tests by running the 'run-ut.sh' script with option `--with-aws`.

Reply:
Could it be ok to check both ec2 and aliyun ? Or you still think using --gtest_filter=-*S3* is good idea?

Reply:
Sorry, I don't get your point. To run these tests successfully, you need to provide `access_key_id` and `secret_access_key`, how can detecting the running environment help?

Reply:
I'll fix by following your idea.